### DISCUSSION TITLE
Execute cronjobs by web request

### DISCUSSION THREAD

**Discussion by zepich** [mosparo/mosparo#174] [Ideas]
If on a hosting, the execution of the cronjobs by CLI command is not possible (disabled at all or because `escapeshellarg` is disabled), or if no cronjobs are available on a hosting, the mosparo cronjobs cannot be executed at all.

The idea is to add a URL to execute the cronjobs (`https://mosparo.example.com/cronjobs`) to solve this issue. The URL could be called from any device (and could also be an external cronjob service).

## Required features

- Add a controller to execute the actions (cleanup database, refresh rulesets, download GeoIP2 database). It needs a (hardcoded) time limit to execute the commands only once per hour (for example) to prevent server overload.
- Add settings to enable the cronjob URL in general and to generate a unique key (for simple URL security).

## Optional features

- Add a setting to control the time limit, after which the cronjob can be executed again.
- Add an option to limit access to the cronjob URL in the security administration (like for the backend and the API).

## Inspired by

- @mpaglia0

## Origin

Discussion #173

**Comment by zepich** [comment]
Hi @mpaglia0 

Today, we released version 1.2, which contains the web cron jobs. Please let me know if you need any help.

Additionally, the Italian translations for the backend you provided are now in the release. Sorry for the delay!

Kind regards,

zepich

**Comment by mpaglia0** [comment]
Thanks a lot @zepich !
My Mosparo installation has been updated and the option has been activated.
As regards the translation I am late... I know...
I will resume the work as soon as possible in order to close the task.
Untranslated strings were a lot!!
Maurizio

**Reply by zepich** [reply]
Hi @mpaglia0 

Absolutely no problem! I'm happy that it's working!

Regarding the translations, that's absolutely fine. Please do not feel stressed or something like that! We're always happy to see when someone contributed to the project, for example, with translations. And there is no stress in doing it very fast or something like that. You can do it whenever you have time! 

Yes, I know, there are a lot of strings... :) But you did a lot already!

Thank you very much for your help in making mosparo better!

Kind regards,

zepich
